File: trade-validator-service/src/main/java/com/coding/trade/model/Trade.java
package com.coding.trade.model;

import com.coding.trade.type.*;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import javax.validation.constraints.NotNull;

@Data
public class Trade {
    
    @JsonProperty(required = true)
    @NotNull
    private String customer;

    @JsonProperty(required = true)
    @NotNull
    private CcyPair ccyPair;

    @JsonProperty(required = true)
    @NotNull
    private TradeType type;

    @JsonProperty(required = true)
    @NotNull
    private Direction direction;

    @JsonProperty(required = true)
    @NotNull
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private String tradeDate;

    @JsonProperty(required = true)
    @NotNull
    private float amount1;

    @JsonProperty(required = true)
    @NotNull
    private float amount2;

    @JsonProperty(required = true)
    @NotNull
    private float rate;

    @JsonProperty(required = true)
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private String valueDate;

    @JsonProperty(required = true)
    @NotNull
    private String legalEntity;

    @JsonProperty(required = true)
    @NotNull
    private String trader;

    @JsonProperty
    private Style style;

    @JsonProperty
    private Strategy strategy;

    @JsonProperty
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private String deliveryDate;

    @JsonProperty
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private String expiryDate;

    @JsonProperty
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private String excerciseStartDate;

    @JsonProperty
    private String payCcy;

    @JsonProperty
    private float premium;

    @JsonProperty
    private String premiumCcy;

    @JsonProperty
    private String premiumType;

    @JsonProperty
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private String premiumDate;
}

File: trade-validator-client/src/app/trade.ts
export class Trade{
  public customer: string;
  public ccyPair = 'EURUSD';
  public type = 'Spot';
  public direction = 'BUY';
  public tradeDate: string;
  public amount1: number;
  public amount2: number;
  public rate: number;
  public valueDate: string;
  public legalEntity: string;
  public trader: string;
  public style = 'EUROPEAN';
  public strategy = 'CALL';
  public deliveryDate: string;
  public expiryDate: string;
  public excerciseStartDate: string;
  public payCcy: string;
  public premium: number;
  public premiumCcy: string;
  public premiumType: string;
  public premiumDate: string;
}
